### PR TITLE
chore: improve syncs log messages

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

- Show results in sync operation payload when successful.
Also simplifying the completion message (moving the results and version in the message payload)
Screenshot 2024-09-05 at 15 26 00

- Show error in sync operation payload when failing
Screenshot 2024-09-05 at 15 27 53

- Also showing the error in an action operation payload when failing to be consistent

- Adds a `starting sync` log message to be consistent with the other script executions

- Removing stringification of errors coming from persist. (Persist error response are well structured)

## Issue ticket number and link

https://linear.app/nango/issue/NAN-1524/improve-sync-errors-logs

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is: 

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced logging capabilities across various components, improving error traceability and observability during actions, sync processes, and webhooks.
	- Updated error handling in the UI to provide structured error messages for better clarity.

- **Bug Fixes**
	- Improved error handling in API interactions to deliver more specific error messages.

- **Documentation**
	- Enhanced logging context in the application to aid in debugging and monitoring.

### CODE REVIEW

File: packages/jobs/lib/execution/action.ts [hunk 1]
```diff
@@ -46,6 +46,12 @@ export async function startAction(task: TaskAction): Promise<Result<void>> {
         }
 
         const logCtx = await logContextGetter.get({ id: String(task.activityLogId) });
+        await logCtx.info(`Starting action '${task.actionName}'`, {
+            input: task.input,
+            action: task.actionName,
+            connection: task.connection.connection_id,
+            integration: task.connection.provider_config_key
+        });
```
Comment (TBonnin) [inline packages/jobs/lib/execution/action.ts:54]: moving the logging from orchestrator.ts to the action handler. Doing the same below for webhook and post connection script

File: packages/jobs/lib/execution/sync.ts [hunk 2]
```diff
@@ -200,6 +201,12 @@ export async function handleSyncSuccess({ nangoProps }: { nangoProps: NangoProps
             environment_id: nangoProps.environmentId,
             provider_config_key: nangoProps.providerConfigKey
         };
+
+        const syncPayload = {
+            scriptVersion: nangoProps.syncConfig.version || 'unknown',
```
Comment (TBonnin) [inline packages/jobs/lib/execution/sync.ts]: arguably the script version should probably be displayed in the UI. Maybe something that we can add later

Comment (bodinsamuel) [reply packages/jobs/lib/execution/sync.ts]: yep indeed, it wasn't a thing when I created the schema but we could add `syncVersion`

File: packages/jobs/lib/execution/sync.ts [hunk 3]
```diff
@@ -589,7 +596,12 @@ async function onFailure({
 
     await updateSyncJobStatus(syncJobId, SyncStatus.STOPPED);
 
-    await logCtx.error(error.message, { error });
+    await logCtx.enrichOperation({
+        error,
+        meta: {
+            scriptVersion: version
```
Comment (bodinsamuel) [inline packages/jobs/lib/execution/sync.ts]: we don't have this information when we create the operation?

File: packages/jobs/lib/execution/sync.ts [hunk 4]
```diff
@@ -385,6 +387,7 @@ export async function handleSyncSuccess({ nangoProps }: { nangoProps: NangoProps
             runTime: (new Date().getTime() - nangoProps.startedAt.getTime()) / 1000,
             failureSource: ErrorSourceEnum.CUSTOMER,
             isCancel: false,
+            version: nangoProps.syncConfig.version,
```
Comment (bodinsamuel) [inline packages/jobs/lib/execution/sync.ts]: at this stage, maybe we should just pass `nangoProps` instead

Comment (TBonnin) [reply packages/jobs/lib/execution/sync.ts]: ideally yes but there are paths where things failed before the nangoProps is instanciated (ie: exception thrown early in startSync)

File: packages/jobs/lib/execution/sync.ts (hunk 3 above)
Comment (TBonnin) [reply packages/jobs/lib/execution/sync.ts]: good point. addressed in [2377384](https://github.com/NangoHQ/nango/pull/2687/commits/2377384c6ceb502f835cfc42d088f596ecd7fa34)

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**